modules/oe_whitelabel_list_pages/tests/src/Functional/ListPagesTest.php

```diff
@@ -72,36 +109,80 @@ public function testListPage(): void {
     $list_page->get('emr_entity_metas')->attach($list_page_entity_meta);
     $list_page->save();
 
-    $this->drupalGet('node/' . $list_page->id());
+    return $list_page;
+  }
 
-    // Assert the left column.
-    $assert_session->elementExists('css', 'div.row > div.col-12.col-lg-3');
-    // Assert offcanvas.
-    $assert_session->elementExists('css', 'div.bcl-offcanvas');
-    $assert_session->elementTextEquals('css', 'h4.offcanvas-title', 'Filter options');
-    $assert_session->elementExists('css', 'button.btn-light > svg');
-    $assert_session->elementTextEquals('css', 'button[data-bs-toggle="offcanvas"]', 'Filters');
-    // Assert Filters.
-    $assert_session->elementExists('css', 'input[name="oe_sc_news_title"]');
-    $assert_session->elementExists('css', 'input[data-drupal-selector="edit-submit"]');
-    $assert_session->elementExists('css', 'input[data-drupal-selector="edit-reset"]');
+  /**
+   * Asserts the title in the facets summary.
+   *
+   * @param int $expected_count
+   *   Expected number of results to be reported in the title.
+   * @param \Behat\Mink\Element\ElementInterface $container
+   *   Container with the expected elements.
+   */
+  protected function assertFacetsSummaryTitle(int $expected_count, ElementInterface $container): void {
+    $title = $container->find('css', 'div.col-md-6.col-lg-8 h4');
+    $text = $title->find('css', 'span.text-capitalize');
+
+    $this->assertSame(
+      sprintf('%s (%s)', $text->getText(), $expected_count),
+      $title->getText()
+    );
+  }
 
-    // Assert right column.
-    $assert_session->elementExists('css', 'div.row > div.col-12.col-lg-9.col-xxl-8');
-    $assert_session->elementContains('css', 'h4.mb-0 > span', 'News list page');
-    $assert_session->elementContains('css', 'h4.mb-0', '(12)');
-    // Assert listing.
-    $assert_session->elementsCount('css', 'hr', 2);
-    $assert_session->elementsCount('css', 'div.listing-item', '10');
-    // Assert pagination.
-    $assert_session->elementExists('css', 'nav > ul.pagination');
-    $assert_session->elementsCount('css', 'ul.pagination > li.page-item', 4);
-
-    // Assert search.
-    $page->fillField('Title', 'News number 8');
-    $page->pressButton('Search');
-    $assert_session->elementContains('css', 'h4.mb-0', '(1)');
-    $assert_session->elementTextEquals('css', 'span.badge.bg-light', 'News number 8');
+  /**
+   * Asserts badges for active filters.
+   *
+   * @param string[] $expected
+   *   Expected badge labels.
+   * @param \Behat\Mink\Element\ElementInterface $container
+   *   Container with the expected elements.
+   */
+  protected function assertActiveFilterBadges(array $expected, ElementInterface $container): void {
+    $badges = $container->findAll('css', '.badge');
+    $this->assertElementsTexts($expected, $badges);
+  }
```

Comment:
Would be nice if we could verify that the wrapper div is gone if no active filters.
```suggestion
  protected function assertActiveFilterBadges(array $expected, ElementInterface $container): void {
    $active_filters_container = $container->find('css', 'div.col-md-12.order-md-2.mt-3');
    if ($expected === []) {
      $this->assertNull($active_filters_container);
      return;
    }
    $this->assertNotNull($active_filters_container);
    $badges = $active_filters_container->findAll('css', '.badge');
    $this->assertElementsTexts($expected, $badges);
  }
```
I don't really like the selectors we need here, but I can't think of anything better - unless we want to add more classes just for the test.